hyp3_sdk/jobs.py

```diff
@@ -170,18 +170,23 @@ def __len__(self):
     def __contains__(self, job: Job):
         return job in self.jobs
 
-    def __delitem__(self, job: Job):
+    def __eq__(self, other: 'Batch'):
+        return self.jobs == other.jobs
+
+    def __delitem__(self, job: int):
         self.jobs.pop(job)
         return self
 
     def __getitem__(self, index: int):
+        if isinstance(index, slice):
+            return Batch(self.jobs[index])
         return self.jobs[index]
 
     def __setitem__(self, index: int, job: Job):
         self.jobs[index] = job
         return self
 
-    def __reverse__(self):
+    def __reversed__(self):
```

Comment:
I actually wonder if it's even worth providing this method?  [From the docs](https://docs.python.org/3/reference/datamodel.html#object.__reversed__)

> If the __reversed__() method is not provided, the reversed() built-in will fall back to using the sequence protocol (__len__() and __getitem__()). Objects that support the sequence protocol should only provide __reversed__() if they can provide an implementation that is more efficient than the one provided by reversed().

`tests/test_jobs.test_reverse` still passes with with this method removed.